File: MonoScripting/Api/EngineClass/EntityComponents/AnimatorComponent.cs
using System;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;

public class AnimatorComponent : IEntityComponent {
    public AnimatorComponent(ulong handle) : base(handle)
    {
    }

    public AnimatorComponent() : base(0)
    {
    }

    public void Play(){
        i_Play(handle);
    }
    public void Stop(){
        i_Stop(handle);
    }

    bool IsPlaying(){
        return i_IsPlaying(handle);
    }

    void AddAnimation(){
        i_AddAnimation(handle);
    }

    void SetAnimationCoeff(uint anim_index, float factor){
        i_SetAnimationCoeff(handle, anim_index, factor);
    }

    void SetAnimation(uint anim_index, Resource anim_resource){
        i_SetAnimation(handle, anim_index, anim_resource);
    }

    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern void i_Play(ulong handle);
    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern void i_Stop(ulong handle);
    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern bool i_IsPlaying(ulong handle);

    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern void i_AddAnimation(ulong handle);

    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern void i_SetAnimationCoeff(ulong handle, uint anim_index, float factor);

    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern void i_SetAnimation(ulong handle, uint anim_index, Resource anim_resource);
}

File: Engine/MonoScripting/Api/EntityComponents/AnimatorComponent.cpp
#include "../ApiBindings.hpp"
#include <Scene/EntityComponents/AnimatorComponent.hpp>
#include <MonoScripting/MonoScriptingLayer.hpp>

using namespace VSGE;

static void Play(AnimatorComponent* anim) {
	anim->Play();
}

static void Stop(AnimatorComponent* anim) {
	anim->Stop();
}

static bool IsPlaying(AnimatorComponent* anim) {
	return anim->IsPlaying();
}

static void AddAnimation(AnimatorComponent* anim) {
	anim->AddNewAnimation();
}

static void SetAnimationCoeff(AnimatorComponent* anim, uint32 index, float coeff) {
	std::vector<AnimationCoeff>& coeffs = anim->GetAnimations();
	coeffs[index].coeff = coeff;
}

static void SetAnimation(AnimatorComponent* anim, uint32 index, Resource* res) {
	std::vector<AnimationCoeff>& coeffs = anim->GetAnimations();
	coeffs[index]._animResource.SetResource(res->GetName());
}

void VSGE::BindAnimator() {
	MonoScriptingLayer::AddInternalCall("AnimatorComponent::i_Play(ulong)", Play);
	MonoScriptingLayer::AddInternalCall("AnimatorComponent::i_Stop(ulong)", Stop);
	MonoScriptingLayer::AddInternalCall("AnimatorComponent::i_IsPlaying(ulong)", IsPlaying);

	MonoScriptingLayer::AddInternalCall("AnimatorComponent::i_AddAnimation(ulong)", AddAnimation);
	MonoScriptingLayer::AddInternalCall("AnimatorComponent::i_SetAnimationCoeff(ulong,single)", SetAnimationCoeff);
	MonoScriptingLayer::AddInternalCall("AnimatorComponent::i_SetAnimation(ulong,uint,string)", SetAnimation);
}